gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/channel/HttpChannelizerIntegrateTest.java

```diff
@@ -18,15 +18,40 @@
  */
 package org.apache.tinkerpop.gremlin.server.channel;
 
-
+import org.apache.http.NoHttpResponseException;
 import org.apache.tinkerpop.gremlin.server.auth.SimpleAuthenticator;
 import org.apache.tinkerpop.gremlin.server.Settings;
 
+import org.junit.Test;
+
 import java.util.Map;
 import java.util.HashMap;
+import java.net.SocketException;
 
 public class HttpChannelizerIntegrateTest extends AbstractGremlinServerChannelizerIntegrateTest {
 
+    @Override
+    public Settings overrideSettings(final Settings settings) {
+        super.overrideSettings(settings);
+        final String nameOfTest = name.getMethodName();
+        if (nameOfTest.equals("shouldBreakOnInvalidAuthenticationHandler") ) {
+            settings.authentication = getAuthSettings();
+            settings.authentication.authenticationHandler = "Foo.class";
+        }
+        return settings;
+    }
+    
+    @Test
+    public void shouldBreakOnInvalidAuthenticationHandler() throws Exception {
+        final CombinedTestClient client =  new CombinedTestClient(getProtocol());
+        try {
+            client.sendAndAssert("2+2", 4);
```

Comment:
this line should be followed by:

```java
fail("Should throw exception");
```

like we don't expect that `sendAndAssert` to actually work, right?

Reply:
Fixed